Sources/BraintreePayPal/BTPayPalCreditFinancing.swift

```diff
@@ -0,0 +1,39 @@
+import Foundation
+
+/// Contains information about a PayPal credit financing option
+@objcMembers public class BTPayPalCreditFinancing: NSObject {
+    /// Indicates whether the card amount is editable after payer's acceptance on PayPal side.
+    public let cardAmountImmutable: Bool
+    
+    /// Estimated amount per month that the customer will need to pay including fees and interest.
+    public let monthlyPayment: BTPayPalCreditFinancingAmount?
+    
+    /// Status of whether the customer ultimately was approved for and chose to make the payment using the approved installment credit.
+    public let payerAcceptance: Bool
+    
+    /// Length of financing terms in months.
+    public let term: Int
+    
+    /// Estimated total payment amount including interest and fees the user will pay during the lifetime of the loan.
+    public let totalCost: BTPayPalCreditFinancingAmount?
+    
+    /// Estimated interest or fees amount the payer will have to pay during the lifetime of the loan.
+    public let totalInterest: BTPayPalCreditFinancingAmount?
```

Comment:
We're not sure how to approach optionality for `monthlyPayment`, `totalCost` and `totalInterest`. [The tests](https://github.com/braintree/braintree_ios/blob/69f7caa0460d43595fc3b5c67cd7fc9fb5d73130/UnitTests/BraintreePayPalTests/BTPaymentMethodNonceParser_PayPal_Tests.swift#L69) expect the type to be optional, but asserts the value not to be `nil`. The [Android tests](https://github.com/braintree/braintree_android/blob/fca6c8ceae54d0f318bc23ad11d519c26a46946f/PayPal/src/test/java/com/braintreepayments/api/PayPalCreditFinancingUnitTest.java#L27) assert that the value should be `null` if the containing JSON object is `null`. A cursory search of the gateway repo for "monthlyPayment", "montly_payment", etc, shows test expectations that the value not be `null`. 

How do we want to handle the case where the gateway unexpectedly returns `null` for one of these properties? Crash the app or set it to `nil`?

How do others think we should handle this?

Reply:
> How do we want to handle the case where the gateway unexpectedly returns null for one of these properties? Crash the app or set it to nil?

I'm a little confused by the question. Are you asking - do we need to handle the `null` case from the gateway at all, since the searching you did confirms that we'll never get a `null` value back?

> Crash the app or set it to nil?
We don't want to crash the merchant app if we unexpectedly get `null`. Our SDK shouldn't crash the app ever.

Reply:
Yeah - I guess my question is: how essential are these to the merchant developer? My instinct is to handle a broken contract from the gateway when we parse - return the whole object as nil if essential properties are missing. The case is rare so it makes sense to me not to take the burden of optionalality off the merchant dev in these cases.

But I also agree with Jax, we should keep the current behavior but discuss a longer term strategy for situations like this once everything is in Swift.